asposeimagingcloud/rest.py

```diff
@@ -0,0 +1,341 @@
+
+    def head(self, url, headers=None, query_params=None, _preload_content=True,
+             _request_timeout=None):
+        return self.request("HEAD", url,
+                            headers=headers,
+                            _preload_content=_preload_content,
+                            _request_timeout=_request_timeout,
+                            query_params=query_params)
+
+    def options(self, url, headers=None, query_params=None, post_params=None,
+                body=None, _preload_content=True, _request_timeout=None):
+        return self.request("OPTIONS", url,
+                            headers=headers,
+                            query_params=query_params,
+                            post_params=post_params,
+                            _preload_content=_preload_content,
+                            _request_timeout=_request_timeout,
+                            body=body)
+
+    def delete(self, url, headers=None, query_params=None, body=None,
+               _preload_content=True, _request_timeout=None):
+        return self.request("DELETE", url,
+                            headers=headers,
+                            query_params=query_params,
+                            _preload_content=_preload_content,
+                            _request_timeout=_request_timeout,
+                            body=body)
+
+    def post(self, url, headers=None, query_params=None, post_params=None,
+             body=None, _preload_content=True, _request_timeout=None):
+        return self.request("POST", url,
+                            headers=headers,
+                            query_params=query_params,
+                            post_params=post_params,
+                            _preload_content=_preload_content,
+                            _request_timeout=_request_timeout,
+                            body=body)
+
+    def put(self, url, headers=None, query_params=None, post_params=None,
+            body=None, _preload_content=True, _request_timeout=None):
+        return self.request("PUT", url,
+                            headers=headers,
+                            query_params=query_params,
+                            post_params=post_params,
+                            _preload_content=_preload_content,
+                            _request_timeout=_request_timeout,
+                            body=body)
+
+    def patch(self, url, headers=None, query_params=None, post_params=None,
+              body=None, _preload_content=True, _request_timeout=None):
+        return self.request("PATCH", url,
+                            headers=headers,
+                            query_params=query_params,
+                            post_params=post_params,
+                            _preload_content=_preload_content,
+                            _request_timeout=_request_timeout,
+                            body=body)
+
+
+class ApiException(Exception):
```

Comment:
@s-zubov Please, refactor the _ApiException_ class and exception handling example, looking at other SDKs which include common error format data, and using the [existing model](https://github.com/aspose-imaging-cloud/aspose-imaging-cloud-python/blob/IMAGINGCLOUD-216/asposeimagingcloud/models/error.py) for this. Please, make the SDK fail and ensure that the appropriate exception is raised and the common error response data is properly deserialized. Thanks.

Reply:
Thank you, I've refactored exception handling. It is a bit tricky, in Python SDK no third-party library is used to deserialize responses from the server. I've rearranged the code that carries deserialization so it is accessible from exception processing function. 

This is an example of an error occurred not on the cloud servers (so, no internal data provided):
> ('502 Error connecting to the API https://api-qa.aspose.cloud/v3.0/imaging/storage/folder/ImagingIntegrationTestData; Message: Bad Gateway', 502, None)

I'll provide an example with captured server error later.

Reply:
This is an example of an error occurred on the server:

> ('404 Error connecting to the API https://api-qa.aspose.cloud/v3.0/imaging/d23d2d/bmp; Message: Not Found', 404, {'code': 'errorItemNotFound',
 'description': 'Operation Failed. Item Not Found.',
 'inner_error': {'_date': datetime.datetime(2019, 7, 19, 7, 23, 2, 991837, tzinfo=tzutc()),
                 'request_id': None},
 'message': "Item doesn't exist at path ImagingCloudTestPython_szubov/d23d2d."})